node/src/server-modules/GlideFt.ts

```diff
@@ -269,6 +269,124 @@ export class GlideFt {
+     * console.log(result); // Output:
+     * // {
+     * //   2,
+     * //   [
+     * //     {
+     * //       "key": "json:2",
+     * //       "value":[
+     * //         {
+     * //           "key": "$",
+     * //           "value": "{\"vec\":[1.1,1.2,1.3,1.4,1.5,1.6]}",
+     * //         },
+     * //         {
+     * //           "key": "__VEC_score",
+     * //           "value": "11.1100006104",
+     * //         }
+     * //       ]
+     * //     },
+     * //     {
+     * //       "key": "json:0",
+     * //       "value":[
+     * //         {
+     * //           "key": "$",
+     * //           "value": "{\"vec\":[1,2,3,4,5,6]}",
+     * //         },
+     * //         {
+     * //           "key": "__VEC_score",
+     * //           "value": "91",
+     * //         }
+     * //       ]
+     * //     }
+     * //   ]
+     * // }
+     * ```
+     */
+    static async search(
+        client: GlideClient | GlideClusterClient,
+        indexName: GlideString,
+        query: GlideString,
+        options?: FTSearchOptions & DecoderOption,
+    ): Promise<
+        (number | GlideRecord<GlideString | GlideRecord<GlideString>>)[]
+    > {
+        const args: GlideString[] = ["FT.SEARCH", indexName, query];
+
+        if (options) {
+            // RETURN
+            if (options.returnFields) {
+                const returnFields: GlideString[] = [];
+                options.returnFields.forEach((returnField) =>
+                    returnField.alias
+                        ? returnFields.push(
+                              returnField.fieldIdentifier,
+                              "AS",
+                              returnField.alias,
+                          )
+                        : returnFields.push(returnField.fieldIdentifier),
+                );
+                args.push(
+                    "RETURN",
+                    returnFields.length.toString(),
+                    ...returnFields,
+                );
+            }
+
+            // TIMEOUT
+            if (options.timeout) {
+                args.push("TIMEOUT", options.timeout.toString());
+            }
+
+            // PARAMS
+            if (options.params) {
+                args.push("PARAMS", (options.params.length * 2).toString());
+                options.params.forEach((param) =>
+                    args.push(param.key, param.value),
+                );
+            }
+
+            // LIMIT
+            if (options.limit) {
+                args.push(
+                    "LIMIT",
+                    options.limit.offset.toString(),
+                    options.limit.count.toString(),
+                );
+            }
+
+            // COUNT
+            if (options.count) {
+                args.push("COUNT");
+            }
+        }
+
+        return _handleCustomCommand(client, args, options) as Promise<
+            (number | GlideRecord<GlideString | GlideRecord<GlideString>>)[]
```

Comment:
Maybe create a separate interface/class for this return type, 
export interface FtSearchResponse = [number, GlideRecord<GlideRecord<GlideString>>]

Reply:
This is fine